hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/validation/OMLayoutVersionValidator.java

```diff
@@ -0,0 +1,105 @@
+ * {@link ValidatorRegistry} class during the initialization of the server.
+ *
+ * The conditions specify the specific use case in which the validator should be
+ * applied to the request. See {@link VersionExtractor} for getting all the supported different
+ * {@link org.apache.hadoop.ozone.Versioned}.
+ * The validator method should be applied to just one specific request type
+ * to help keep these methods simple and straightforward. If you want to use
+ * the same validation for different request types, use inheritance, and
+ * annotate the override method that just calls super.
+ * Note that the aim is to have these validators together with the request
+ * processing code, so the handling of these specific situations are easy to
+ * find.
+ *
+ * The annotated methods have to have a fixed signature.
+ * A {@link RequestProcessingPhase#PRE_PROCESS} phase method is running before
+ * the request is processed by the regular code.
+ * Its signature has to be the following:
+ * - it has to be static and idempotent
+ * - it has to have two parameters
+ * - the first parameter it is an
+ * {@link
+ * org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMRequest}
+ * - the second parameter of type {@link ValidationContext}
+ * - the method has to return the modified request, or throw a ServiceException
+ *   in case the request is considered to be invalid
+ * - the method does not need to care about preserving the request it gets,
+ *   the original request is captured and saved by the calling environment.
+ *
+ * A {@link RequestProcessingPhase#POST_PROCESS} phase method is running once
+ * the
+ * {@link
+ * org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.OMResponse}
+ * is calculated for a given request.
+ * Its signature has to be the following:
+ * - it has to be static and idempotent
+ * - it has three parameters
+ * - similalry to the pre-processing validators, first parameter is the
+ *   OMRequest, the second parameter is the OMResponse, and the third
+ *   parameter is a ValidationContext.
+ * - the method has to return the modified OMResponse or throw a
+ *   ServiceException if the request is considered invalid based on response.
+ * - the method gets the request object that was supplied for the general
+ *   request processing code, not the original request, while it gets a copy
+ *   of the original response object provided by the general request processing
+ *   code.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+@RegisterValidator
+public @interface OMLayoutVersionValidator {
+
+  /**
+   * Defines if the validation has to run before or after the general request
+   * processing.
+   * @return if this is a pre or post processing validator
+   */
+  RequestProcessingPhase processingPhase();
+
+  /**
+   * The type of the request handled by this validator method.
+   * @return the requestType to whihc the validator should be applied
```

Comment:
```suggestion
   * The type(s) of the request(s) handled by this validator method.
   * In case you are considering to specify multiple request types, please read the note above on
   * using multiple request types with this kind of validator method.
   * @return the requestType(s) onto which the validator should be applied
```